app/src/main/java/org/thoughtcrime/securesms/notifications/DefaultMessageNotifier.java

```diff
@@ -505,24 +511,45 @@ private NotificationState constructNotificationState(@NonNull  Context context,
           continue;
         }
       }
+
+      // If this is a message request from an unknown user..
       if (messageRequest) {
         body = SpanUtil.italic(context.getString(R.string.message_requests_notification));
+
+      // If we received some manner of notification but Session is locked..
       } else if (KeyCachingService.isLocked(context)) {
         body = SpanUtil.italic(context.getString(R.string.MessageNotifier_locked_message));
+
+      // ----- All further cases assume we know the contact and that Session isn't locked -----
+
+      // If this is a notification about a multimedia message from a contact we know about..
       } else if (record.isMms() && !((MmsMessageRecord) record).getSharedContacts().isEmpty()) {
         Contact contact = ((MmsMessageRecord) record).getSharedContacts().get(0);
         body = ContactUtil.getStringSummary(context, contact);
+
+      // If this is a notification about a multimedia message which contains no text but DOES contain a slide deck with at least one slide..
       } else if (record.isMms() && TextUtils.isEmpty(body) && !((MmsMessageRecord) record).getSlideDeck().getSlides().isEmpty()) {
         slideDeck = ((MediaMmsMessageRecord)record).getSlideDeck();
         body = SpanUtil.italic(slideDeck.getBody());
+
+      // If this is a notification about a multimedia message, but it's NOT (perhaps because it's a `MediaMmsMessageRecord` that for some
+      // reason returns false), AND it contains a slide deck with at least one slide..
       } else if (record.isMms() && !record.isMmsNotification() && !((MmsMessageRecord) record).getSlideDeck().getSlides().isEmpty()) {
         slideDeck = ((MediaMmsMessageRecord)record).getSlideDeck();
         String message      = slideDeck.getBody() + ": " + record.getBody();
         int    italicLength = message.length() - body.length();
         body = SpanUtil.italic(message, italicLength);
+
+      // If this is a notification about an invitation to a community..
       } else if (record.isOpenGroupInvitation()) {
         body = SpanUtil.italic(context.getString(R.string.ThreadRecord_open_group_invitation));
       }
+      else if (record.isMmsNotification()) {
+        // If this is a Mms notification but any of the above spaghetti logic checks weren't met..
+      } else {
+        // We will hit this else block if the notification is for a text only / SMS message from a known contact
+      }
```

Comment:
Is there a reason for these empty blocks?

Reply:
Removed.